---
title: "Univariate GWAS"
subtitle: "STAT 892-004 Integrative Data Science for Plant Phenomics"
author: "Gota Morota"
date: "2018/02/06"
output:
  xaringan::moon_reader:
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
---

```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE)
```



class: inverse, center, middle

# Project


---
# Project teams 
Who will be my project partner?

- We will ask R to decide 

- 8 students from Agronomy & Horticulture Department

- 7 students from other Departments 

- teams.R R code will randomly assign you to someone from different Department

- Proposal due on Tuesday, February 27

- Final report due on Monday, April 16


---
class: inverse, center, middle

# Schedule



---
# Change of schedule 

- Tuesday, February 6 - Genomics
 
- Thursday, February 8 - Genomics 
    
- Tuesday, February 13 - High-throughput phenotyping

- Thursday, February 15 - High-throughput phenotyping  

- Tuesday, February 20 - Genomics

- Thursday, February 22 - Genomics 


---
class: inverse, center, middle

# Introduction


---
# Prediction vs. Inference 
Complex traits are controlled by large number of genes with small effects, and  influenced by both genetics and environments

- Inference (location)
    - average effects of allele substitution

- Inference (variability)
   - variance component estimation
   - genomic heritability
  
Combination of above two (e.g., estimate proportion of additive genetic variance explained by QTLs)

- Prediction 
    - genomic selection
    - prediction of yet-to-be observed phenotypes



---
# Prediction vs. Inference 
<div align="center">
<img src="Lo2015PNAS.png" width=900 height=400>
</div>

* [http://www.pnas.org/content/112/45/13892.abstract
](http://www.pnas.org/content/112/45/13892.abstract
)



---
# How to parameterize  response variable y?

- Prediction of additive genetic effects 
     - $\mathbf{ y =  E + a + \boldsymbol{\epsilon}}$ 
     
- Prediction of total genetic effects **parametrically**
     - $\mathbf{ y =  \mathbf{E} + \underbrace{\mathbf{ a + d + a*a + a*d + d*d}}_{g} + } \boldsymbol{\epsilon}$ 
     
- Prediction of total genetic effects **non-parametrically**
     - $\mathbf{ y =  \mathbf{E} + \mathbf{g} + \boldsymbol{\epsilon}}$ 


---
# Pixelomics
![](plant_01.png)
&nbsp;
&nbsp;
![](plant_02.jpg)

<center>
&darr;
<br/>
<br/>
Converting image data into <a herf="https://en.wikipedia.org/wiki/Numerical_digit"> numerical values</a>
</center>





---
# Genomic information (e.g., SNPs)
![](SNPs.png)

.center[Repeat of numbers 0, 1, and 2]

---
# Quantitative genetics
Connecting image data with genomic information 
<center>
<div>
<img src="plant_01.png" width=100 height=100>
=
<img src="SNPs.png" width=100 height=100> + error 
</div>
</center>
This is equivalent to 
\begin{align*}
\mathbf{y} &= \mathbf{W}\mathbf{a} + \boldsymbol{\epsilon} \\
\underbrace{\begin{bmatrix}    
y_1\\
y_2\\
\vdots \\
y_n\end{bmatrix}}_{n \times 1}  
&= \underbrace{\begin{bmatrix}
   w_{11} & w_{12} & \cdots & w_{1m} \\
    w_{21} & w_{22} & \cdots & w_{2m} \\
    \vdots & \vdots & \ddots & \vdots \\
    w_{n1} & w_{n2} & \cdots & w_{nm}
\end{bmatrix}}_{n \times m} \quad
\underbrace{\begin{bmatrix}
a_1\\
a_2\\
\vdots \\
a_m\end{bmatrix}}_{m \times 1} 
+\underbrace{\begin{bmatrix}
\epsilon_1\\
\epsilon_2\\
\vdots \\
\epsilon_m\end{bmatrix}}_{n \times 1} 
\end{align*}

where $n$ is the number of individuals (e.g., accessions) and $m$ is the number of SNPs. 




---
# Genetic values
Quantitative genetic model:

\begin{align*}
\mathbf{y} &= \mathbf{g} + \boldsymbol{\epsilon} \\
\end{align*}
where $\mathbf{y}$ is the vector of observed phenotypes, $\mathbf{g}$ is the vector of genetic values, and $\boldsymbol{\epsilon}$ is the vector of residuals. 

Example:

| Plant ID   | y           | g  |   e  |
| ------------- |:-------------:| -----:|------|
| 1      | 10 | ? |  ?     |
| 2      |  7     |   ? |  ?    |
| 3 | 12     |    ? |    ? |




---
# Genetic values
Quantitative genetic model:

\begin{align*}
\mathbf{y} &= \mathbf{g} + \boldsymbol{\epsilon} \\
\end{align*}
where $\mathbf{y}$ is the vector of observed phenotypes, $\mathbf{g}$ is the vector of genetic values, and $\boldsymbol{\epsilon}$ is the vector of residuals. 

Example:

| Plant ID    | y           | g  |   e  |
| ------------- |:-------------:| -----:|------|
| 1      | 10 | 5 |  5     |
| 2      |  7     |   6 |  1    |
| 3 | 12     |    2 |    10 |


--

We approximate unknown $\mathbf{g}$ with $\mathbf{Wa}$. 

\begin{align*}
\mathbf{y} &= \mathbf{g} + \boldsymbol{\epsilon} \\
 &\approx \mathbf{W}\mathbf{a} + \boldsymbol{\epsilon} 
\end{align*}


---
# Assumption
We assume that image data are capturing some aspects of real biological signal of plants. 

If this is assumption is not met..
--
<center>
<iframe src="https://giphy.com/embed/lKZEeXJGhU1d6" width="250" height="150" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/scared-despicable-me-lKZEeXJGhU1d6">via GIPHY</a>
</p>
</center>

Close communication among other disciplines (e.g., breeding, physiology, and computer science) are essential!



---
# What is unique about image data?
.pull-left[
<div align="left">
<iframe src="https://innovate.unl.edu/video/leasing-options/greenhouse-innovation-center.mp4" width="250" height="150" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://innovate.unl.edu/greenhouse-innovation-center">UNL Greenhouse Innovation Center</a>
</p>
</div>

<div align="left">
<iframe width="260" height="200" src="https://www.youtube.com/embed/wor4BFjbIyI?rel=0" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
<p><a href="https://www.youtube.com/watch?v=wor4BFjbIyI">Spidercam</a>
</div>

]

.pull-right[
<div align="right">
<img src="https://www.frontiersin.org/files/Articles/254051/fpls-08-00421-HTML/image_m/fpls-08-00421-g002.jpg" width=400 height=400><p>Unmanned aerial vehicle<a href="https://www.frontiersin.org/articles/10.3389/fpls.2017.00421/full"> (Watanabe et al. 2017)</a>
</div>
]


---
# Longitudinal data
Projected shoot area (PSA) = the sum of plant pixel from two side-view images and one top-view 
<div align="center">
<img src="mackPSA.png" width=400 height=200><p><a href=""> Campbell et al. (In prep)</a>
</div>
- Single time point analysis
   - cross-sectional analysis
- Longitudinal analysis 
   - leverage covariance among time points
   - account for longitudinal curve
   
  

---
# Expectation and variance 
Define the random variable $W$ which counts the number of reference allele $A$.
\begin{align*}
  W &=
  \begin{cases}
    2 & \text{if } AA  \text{ with frequency } p^2 \\
    1 & \text{if } Aa \text{ with frequency } 2p(1-p) \\
    0 & \text{if } aa \text{ with frequency } (1-p)^2
  \end{cases} \\
\end{align*}
where $p$ is the allele frequency of $A$. 

Then, 
\begin{align*}
E[W] &= 0 \times (1 - p_j)^2 + 1 \times [2p(1-p)] + 2 \times p^2 \\
&= 2p \\
E[W^2] &= 0^2 \times (1 - p_j)^2 + 1^2 \times [2p(1-p)] + 2^2 \times p^2 \\
&= 2p(1-p) + 4p^2 \\
\end{align*}
Thus, the variance of allelic counts is 
\begin{align*}
Var(W) &= E[W^2] - E[W]^2 \\
&= 2p(1-p) + 4p^2  - 4p^2\\
&= 2p(1-p)
\end{align*}


---
class: inverse, center, middle

# Genome-wide association studies (GWAS)


---
# Ordinary least squares (OLS) 
Quantitative genetic model: $\mathbf{y} = \mathbf{Wa} + \boldsymbol{\epsilon}$

How to find the SNP effects ( $\mathbf{a}$ )?

--

We minimize the residual sum of squares
\begin{align*}
\boldsymbol{\epsilon}' \boldsymbol{\epsilon} &= (\mathbf{y-Wa})'(\mathbf{y-Wa}) \\
&= \mathbf{y}'\mathbf{y}  - \mathbf{y}'\mathbf{W} \mathbf{a}- \mathbf{a}'\mathbf{W}'\mathbf{y} + \mathbf{a}'\mathbf{W}'\mathbf{W}\mathbf{a}  \\
&= \mathbf{y}'\mathbf{y} - 2\mathbf{a}'\mathbf{W}'\mathbf{y} + \mathbf{a}'\mathbf{W}'\mathbf{W}\mathbf{a}
\end{align*}

We take a partial derivative with respect to $\mathbf{a}$
\begin{align*}
  \frac{\partial \boldsymbol{\epsilon\epsilon}'}{\partial \boldsymbol{a}} &= 2 \mathbf{W}'\mathbf{W} \mathbf{a} - 2\mathbf{X}'\mathbf{y}
\end{align*}

By setting the equation equal to zero, we obtain a least square estimator of $\mathbf{a}$. 
\begin{align*}
\mathbf{W}'\mathbf{W} \mathbf{a} &= \mathbf{W}' \mathbf{y} \\
\hat{\mathbf{a}} &= (\mathbf{W}'\mathbf{W})^{-1} \mathbf{W}' \mathbf{y}
\end{align*}



---
# Ordinary least squares (OLS) 
- $\hat{\mathbf{a}}$ is the vector of regression coefficient for markers, i.e., effect size of SNPs

- if the Gauss-Markov theorem is met, $E[\hat{\mathbf{a}}] = \mathbf{a} \rightarrow$ BLUE 
    - $E[\boldsymbol{\epsilon}] = 0$,  $Var[\boldsymbol{\epsilon}] = \mathbf{I}\sigma^2_{\epsilon}$
    

What if  number of SNPs ( $m$ ) $>>$  number of individuals ( $n$ ) ???

--

- $(\mathbf{W}'\mathbf{W})^{-1}$ does not exist 

- Effective degrees of freedom 


---
# OLS: Single marker regression
Test each marker for the presence of QTLs and select those with significant effects

Problems: marker effect sizes are exaggerated

Suppose the true model is given by two causal SNPs 
\begin{align*}
  \mathbf{y} & = \mathbf{w}_1a_1 + \mathbf{w}_2a_2 + \boldsymbol{\epsilon} \\
  \mathbf{y} & = \begin{smallmatrix} \underbrace{\mathbf{W}}_{n \times 2}\underbrace{\mathbf{a}}_{2 \times 1} \end{smallmatrix} + \boldsymbol{\epsilon}
\end{align*}

The OLS estimator for the full mdoel is 
\begin{align*}
  \begin{bmatrix}
    \hat{a}_1 \\
    \hat{a}_2 
  \end{bmatrix} &=
    \begin{bmatrix}
      \mathbf{w}'_1\mathbf{w}_1 & \mathbf{w}'_1\mathbf{w}_2 \\
      \mathbf{w}'_2\mathbf{w}_1 & \mathbf{w}'_2\mathbf{w}_2 
    \end{bmatrix}^{-1}
    \begin{bmatrix}
      \mathbf{w}'_1\mathbf{y} \\
      \mathbf{w}'_2\mathbf{y} \\
    \end{bmatrix} \\
    \hat{\mathbf{a}} &= (\mathbf{W'W})^{-1}\mathbf{W}'\mathbf{y}
\end{align*}


---
# OLS: Single marker regression
The expectation of $\hat{\mathbf{a}}$ is
\begin{align*}
  E(\hat{\mathbf{a}} | \mathbf{W}) = (\mathbf{W'W})^{-1}\mathbf{W'}E(\mathbf{y} ) = (\mathbf{W'W})^{-1}\mathbf{W'Wa} = \mathbf{a}
\end{align*}
which is a nice property of BLUE.

Now, what if we fit a single SNP model $\mathbf{y} = \mathbf{w}_1a_1 + \boldsymbol{\epsilon}$?

The OLS estimate is $\hat{a}_1 = (\mathbf{w'_1w_1})^{-1}\mathbf{w}'_1\mathbf{y}$

The expectation of $\hat{a}_1$ is
\begin{align*}
  E(\hat{a}_1 | \mathbf{w}_1) &= (\mathbf{w'_1w_1})^{-1}\mathbf{w}'_1E(\mathbf{y}) \\
  &= (\mathbf{w'_1w_1})^{-1}\mathbf{w'_1}[\mathbf{w_1a_1 + w_2a_2}] \\
  &= \mathbf{(w'_1w_1)}^{-1}\mathbf{w'_1w_1}a_1 + (\mathbf{w'_1w_1})^{-1}\mathbf{w'_1w_2}a_2 \\
  &= a_1 + (\mathbf{w'_1w_1})^{-1}\mathbf{w'_1w_2}a_2 
\end{align*}

- OLS is biased if full model holds but fit a misspecified model
- this bias is proportional to $(\mathbf{w'_1w_1})^{-1}\mathbf{w'_1w_2}a_2$
- the same applies when there are more than two SNPs 



---
class: inverse, center, middle

# Population structure



---
# Principal components (PC)
PC captures population structure

- Menozzi et al. (1978).  [doi:10.1126/science.356262](http://doi.org/10.1126/science.356262)
- Cavalli-Sforza et al. (1996). [ISBN-13: 978-0691029054](http://www.amazon.com/gp/product/0691029059/)

Example in rice diversity panel data
![](riceDiversityPC1-2.png)


---
# Population stratification 
Population structure as a confounder
<div align="center">
<img src="myBalding2006.png" width=700 height=260>
</div>

- Knowler et al. (1988) Gm3;5,13,14 and type 2 diabetes mellitus: an association in American Indians with genetic admixture. Am J Hum Genet. 
- inflation of SNP effect sizes
- inflation of genomic heritability
- overestimation of prediction accuracy 

---
# What are principal components?
$n \times m$ matrix of SNPs ( $\mathbf{W}$ )

- singular value decomposition of $\mathbf{W} = \mathbf{U}\boldsymbol{\Sigma}\mathbf{V}'$

$n \times n$ genomic relationship matrix ( $\mathbf{G}$ )

- eigen decomposition of $\mathbf{G} = \mathbf{U}\mathbf{D}\mathbf{U}'$ 

Principal components

1. $\text{PC} = \mathbf{U}$
2. $\text{PC} = \sqrt{\mathbf{D}}\mathbf{U}$


---
# Genomic relationship matrix 
Recall that
\begin{align*}
\mathbf{y} &= \mathbf{g} + \boldsymbol{\epsilon} = \mathbf{W}\mathbf{a} + \boldsymbol{\epsilon}
\end{align*}

Assume genetic value is parameterized as $g_{i} = \sum w_{ij} a_j$
where both $w$ and $a$ are treated as random and independent. 
Assuming linkage  equilibrium of markers (all loci are mutually independent) 
\begin{align*}
\sigma^2_g &= \sum_j 2 p_j(1-p_j) \cdot \sigma^2_{a_j}. \notag \\
\end{align*}
Under the homogeneous marker variance assumption
\begin{align}
\sigma^2_{a} &= \frac{\sigma^2_g}{2 \sum_j p_j(1-p_j) }.
\end{align}
Then, variance of genetic values is 
\begin{align*}
Var(\mathbf{g}) &= Var(\mathbf{W}\mathbf{a}) = \mathbf{WW'}\sigma^2_{a} \\
&= \frac{\mathbf{WW'}}{2 \sum_j p_j(1-p_j)} \sigma^2_g = \mathbf{G}\sigma^2_g
\end{align*}



---
class: inverse, center, middle

# Linear mixed model 



---
# Mixed model association for GWAS 
Single marker-based mixed model association (MMA)
\begin{align*}
\mathbf{y} &= \mu + \mathbf{w_ja_j} + \mathbf{g} + \boldsymbol{\epsilon} \\
\mathbf{g} &\sim N(0, \mathbf{G}\sigma^2_{g})
\end{align*}

$\mathbf{G}$ captures population structure and polygenic effects


--

Double counting?

--

Alternatively, 
\begin{align*}
\mathbf{y} &= \mu + \mathbf{w_ja_j} + \mathbf{g} + \boldsymbol{\epsilon} \\
\mathbf{g} &\sim N(0, \mathbf{G}_{-k}\sigma^2_{g_{-k}})
\end{align*}
where $-k$ denotes the $k$th chromosome removed


